internal/myks/plugins.go

```diff
@@ -0,0 +1,150 @@
+func FindPluginsInPaths(paths []string, binaryPrefix string) []Plugin {
+	var plugins []Plugin
+	if len(paths) == 0 {
+		return plugins
+	}
+	for _, path := range paths {
+		if len(strings.TrimSpace(path)) == 0 {
+			continue
+		}
+		files, err := os.ReadDir(path)
+		if err != nil {
+			log.Trace().Err(err).Msgf("Unable to read directory %s", path)
+			continue
+		}
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			if !strings.HasPrefix(file.Name(), binaryPrefix) {
+				continue
+			}
+			executable := filepath.Join(path, file.Name())
+			if isExec, err := isExecutable(executable); err != nil {
+				log.Debug().Err(err).Msgf("Unable to check if %s is executable", executable)
+			} else if !isExec {
+				log.Trace().Msgf("Skipping %s because it is not executable", executable)
+				continue
+			}
+			plugins = append(plugins, NewPluginFromCmd(executable, binaryPrefix))
+		}
+	}
+	return plugins
+}
+
+func isExecutable(fullPath string) (bool, error) {
+	info, err := os.Stat(fullPath)
+	if err != nil {
+		return false, err
+	}
+
+	if runtime.GOOS == "windows" {
+		fileExt := strings.ToLower(filepath.Ext(fullPath))
+
+		switch fileExt {
+		case ".bat", ".cmd", ".com", ".exe", ".ps1":
+			return true, nil
+		}
+		return false, nil
+	}
+
+	if m := info.Mode(); !m.IsDir() && m&0o111 != 0 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+type PluginCmd struct {
+	name string
+	cmd  string
+}
+
+func (p PluginCmd) Name() string {
+	return p.name
+}
+
+func (p PluginCmd) Exec(a *Application, args []string) error {
+	step := "Plugin " + p.Name()
+	log.Trace().Msg(a.Msg(step, "execution started"))
+
+	env, err := p.generateEnv(a)
+	if err != nil {
+		log.Error().Err(err).Msg(a.Msg(step, "Generating data values failed"))
+		return err
+	}
+
+	cmd := exec.Command(p.cmd, args...)
+
+	var stdoutBs, stderrBs bytes.Buffer
+	cmd.Stdout = &stdoutBs
+	cmd.Stderr = &stderrBs
+	cmd.Env = append(os.Environ(), mapToSlice(env)...)
+
+	err = cmd.Run()
+	if err != nil {
+		log.Error().Err(err).
+			Str("stderr", stderrBs.String()).
+			Str("stdout", stdoutBs.String()).
+			Msg(a.Msg(step, "Plugin execution failed"))
+	} else {
+		log.Debug().Str("stdout", stdoutBs.String()).
+			Msg(a.Msg(step, "Plugin execution succeeded"))
+	}
+	return err
+}
+
+func (p PluginCmd) generateEnv(a *Application) (map[string]string, error) {
+	env := map[string]string{
+		"MYKS_ENV":              a.e.Id,
+		"MYKS_APP":              a.Name,
+		"MYKS_APP_PROTOTYPE":    a.Prototype,
+		"MYKS_ENV_DIR":          a.e.Dir,
+		"MYKS_ENV_DATA_FILE":    a.e.renderedEnvDataFilePath,
+		"MYKS_RENDERED_APP_DIR": "rendered/envs/" + a.e.Id + "/" + a.Name, // TODO: provide func and use it everywhere,
+		"MYKS_ARGOCD_ENABLED":   strconv.FormatBool(a.argoCDEnabled),
+	}
+	if a.argoCDEnabled {
+		env["MYKS_ARGOCD_APP_NAME"] = a.getArgoCDDestinationDir()
+		env["MYKS_ARGOCD_APP_PROJECT"] = "rendered/argocd/" + a.e.Id + "/" + a.Name // TODO: provide func and use it everywhere,
+	}
```

Comment:
```suggestion
	env := map[string]string{
		"MYKS_ENV":              a.e.Id,
		"MYKS_APP":              a.Name,
		"MYKS_APP_PROTOTYPE":    a.Prototype,
		"MYKS_ENV_DIR":          a.e.Dir,
		"MYKS_RENDERED_APP_DIR": "rendered/envs/" + a.e.Id + "/" + a.Name, // TODO: provide func and use it everywhere
	}
```

Since we now have `MYKS_DATA_VALUES`, most of the info can be retrieved from it. Anything hard to retrieve directly or impossible to retrieve at the moment (like the rendered dir, because it requires knowing the configuration of the `Globe` object), we can add into the data schema as we did with the `context` section. But I think it's convenient to have the most basic things as environment variables. So, I propose to remove ArgoCD-related stuff and `MYKS_ENV_DATA_FILE`, because it probably won't be used much.